framework/src/util/ServiceRegistrationLocks.h

```diff
@@ -0,0 +1,43 @@
+/*=============================================================================
+  Library: CppMicroServices
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+=============================================================================*/
+
+#ifndef CPPMICROSERVICES_SERVICEREGISTRATIONLOCKS_H
+#define CPPMICROSERVICES_SERVICEREGISTRATIONLOCKS_H
+
+#include "ServiceRegistrationBasePrivate.h"
+#include "ServiceRegistrationCoreInfo.h"
+
+namespace cppmicroservices
+{
+
+    /**
+     * \ingroup MicroServices
+     */
+    class ServiceRegistrationLocks
```

Comment:
A comment describing the purpose of this class would be useful.

Reply:
This has been added to this class